### ISSUE TITLE
jmx_exporter 0.17.2 doesn't work on activemq-artemis 2.25.0 and later version

### ISSUE THREAD

**Issue by wureka** [prometheus/jmx_exporter#761]
- ActiveMQ-Artemis:  2.27.1, installed in /opt/apache-artemis-2.27.1. The server IP is 192.168.21.68
- Java version: OpenJDK 11.0.16.
- ActiveMQ Broker path: /opt/broker_2168

I added below command into /opt/broker_2168/bin/artemis as below:

```
exec "$JAVACMD" \
    $JAVA_ARGS \
    -javaagent:$ARTEMIS_HOME/lib/jmx_prometheus_javaagent-0.17.2.jar=9133:$ARTEMIS_INSTANCE_ETC/artemis-jmx-exporter.yml \
    -Dhawtio.role="$HAWTIO_ROLE" \
    ....
```

the above artemis-jmx-exporter.yml is identical to [https://github.com/prometheus/jmx_exporter/blob/main/example_configs/artemis-2.yml](https://github.com/prometheus/jmx_exporter/blob/main/example_configs/artemis-2.yml)

When I use FireFox to visit http://192.168.21.68:9133/metrics , I can see jvm information, but no artemis information.

The same configurations can apply to ActiveMQ-Artemis 2.24.0 and below versions

![image](https://user-images.githubusercontent.com/9668764/209025844-05b7d2fb-fa11-412d-a6cd-03bbc1c5ca5e.png)

**Comment by Botje** [comment]
I traced through ActiveMQ Artemis with a debugger and it is [documented behavior on the Artemis side](https://activemq.apache.org/components/artemis/documentation/latest/management.html#local-jmx-access-with-jconsole) (mentally replace JConsole with jmx-exporter):
> Due to the authorisation which is enabled by default Apache ActiveMQ Artemis can not be managed locally using JConsole when connecting as a local process. This is because JConsole does not pass any authentication information when connecting this way which means the user cannot therefore be authorised for any management operations. In order to use JConsole the user will either have to disable authorisation by completely removing the authorisation element from management.xml or by enabling remote access and providing the proper username and password credentials (discussed next).

The final decision is made inside `org.apache.activemq.artemis.core.server.management.ArtemisMBeanServerGuard#handleInvoke`, where there are two paths:
1. MBeans whose domain is in the management.xml `<allowlist>` element are allowed.
2. Otherwise, the required roles are looked up (see the `<default-access>` and `<role-access>` elements) and the current user's roles are inspected.

However, to the best of my knowledge there is no "current user" if jmx-exporter is running as a Java agent.

For now, I added `<entry domain="org.apache.activemq.artemis" />` to the `<allowlist>` element in Artemis and now I can see all attributes in the output.
